documentation/projects/proposals/trust_and_safety/content_report_moderation/20230517-implementation_plan_mature_copy_change.md

```diff
@@ -0,0 +1,235 @@
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/383)
+- [Project Proposal](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20230411-project_proposal_content_report_moderation.html)
+
+## Overview
+
+<!-- An overview of the implementation plan, if necessary. Save any specific steps for the section(s) below. -->
+
+This implementation plan concerns the copy changes necessary for updating
+references of `mature` content to `sensitive` content. These changes will be
+made to the frontend and Django API, both in code and in web-facing copy. There
+should be no design changes or new features as part of this plan, only changes
+to copy/text itself. Field and model names for the API should also be updated
+when possible, but without the need for database migrations (i.e., using
+[Django's built-in tools for specifying different underlying names for fields and tables](https://docs.djangoproject.com/en/4.2/ref/models/options/#table-names)).
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+All references to `mature` in code and copy will be updated to `sensitive`,
+except where `mature` is
+[part of a data model](https://github.com/WordPress/openverse/blob/0f90f84d4bf856297356a918e4cc4fe42d07ffc2/ingestion_server/ingestion_server/elasticsearch_models.py#L118)
+or makes the most sense to keep (e.g. when interacting with certain provider
+aspects).
+
+## Outlined Steps
+
+<!-- Describe the implementation step necessary for completion. -->
+
+The work is broken up into two services (API and frontend) and those services
+are broken into two aspects (code and copy).
+
+### API
+
+#### Code
+
+The following models will need to be updated. Since we want to avoid a migration
+with this work, they will need to
+[explicitly reference the old table names using `Meta::db_table`](https://docs.djangoproject.com/en/4.2/ref/models/options/#table-names).
+
+- [`AbstractMatureMedia`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/media.py#L334-L333)
+- [`MatureImage`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/image.py#L83)
+- [`MatureAudio`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/audio.py#L263)
+
+The
+[`mature` reason](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/media.py#L22-L21)
+(and
+[`mature_filtered`](https://github.com/WordPress/openverse/blob/2041c5df1e9d5d1f9f37e7c177f2e70f61ea5dba/api/api/models/media.py#L18-L17)
+value) in `AbstractMediaReport` definition will also need to be changed to
+`sensitive` and `sensitive_filtered` respectively. In order to remain backwards
+compatible with the frontend, incoming references on reports for `mature` will
+need to be mapped to the new value.
```

Comment:
If I'm understanding the approach suggested here, this would have us update all the previous report values, right? I think that will require multiple steps including Django migrations (Django insists on encoding "choices" into migrations even though Postgres doesn't know anything about it) and (even if Django migrations aren't needed) a management command to update the old values (as per [our documentation on data transformations](https://docs.openverse.org/general/zero_downtime_database_management.html#how-to-achieve-zero-downtime-deployments)). Is my understanding of the suggestion correct? If so I think it would be worth outlining the steps for that explicitly.

FWIW, I don't think this violates the no-migrations stipulation from the project proposal as that was aimed primarily at table and column name changes rather than _data_ changes. Table and column name changes are just a lot harder to juggle and can be masked in code and UI using Django's tools. The data inside the columns, however, is easier to update and much peskier if we don't (easy to accidentally mess up queries, etc).

Reply:
Oh that's very true, thank you for mentioning it! I'll add that to this IP 🙂